src/main/java/com/byznass/tiolktrack/postgres/dao/PostgresGpsProvider.java

```diff
@@ -33,40 +33,40 @@ public PostgresGpsProvider(Connection connection, LocationProvider locationProvi
 	@Override
 	public Gps getGpsById(String gpsId) throws NoGpsWithIdException, TiolkTrackException {
 
-		try {
-			LOGGER.info("Retrieving Gps entity with id=\"{}\" from postgres database", gpsId);
-			checkIfExists(gpsId);
-			List<Location> locations = locationProvider.getLocationsForGps(gpsId);
+		LOGGER.info("Retrieving Gps entity with id=\'{}\' from postgres database", gpsId);
+		checkIfExists(gpsId);
+		List<Location> locations = locationProvider.getLocationsForGps(gpsId);
+		LOGGER.info("Successfully retrieved Gps entity with id=\'{}\' from postgres database", gpsId);
 
-			return new Gps(gpsId, locations);
-		} catch (SQLException e) {
-			LOGGER.error("Error while retrieving Gps entity with id=\"{}\" from postgres", gpsId, e);
-			throw new TiolkTrackException(String.format("Cannot retrieve Gps with id=\"%s\" from database", gpsId), e);
-		}
+		return new Gps(gpsId, locations);
 	}
 
-	private void checkIfExists(String gpsId) throws SQLException {
+	@Override
+	public boolean exists(String gpsId) throws TiolkTrackException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.kernel.TiolkTrackException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)